r/R/dplyr-functions.R

```diff
@@ -634,20 +634,53 @@ nse_funcs$wday <- function(x, label = FALSE, abbr = TRUE, week_start = getOption
 }
 
 nse_funcs$log <- function(x, base = exp(1)) {
-  
+
   if (base == exp(1)) {
     return(Expression$create("ln_checked", x))
   }
-  
+
   if (base == 2) {
     return(Expression$create("log2_checked", x))
   }
-  
+
   if (base == 10) {
     return(Expression$create("log10_checked", x))
-  } 
+  }
   # ARROW-13345
   stop("`base` values other than exp(1), 2 and 10 not supported in Arrow", call. = FALSE)
 }
 
 nse_funcs$logb <- nse_funcs$log
+
+nse_funcs$if_else <- function(condition, true, false, missing = NULL){
+  # We ought to assert that the types of the true and false conditions will result
+  # in the same types. We can't compare the objects themselves directly because
+  # they might be expressions (that will result in a type) or R objects that will
+  # need to be compared to see if they are compatible with arrow types.
+  # ARROW-13186 might make this easier with a more robust way.
+  # TODO: do this ^^^
+
+  # if_else only supports boolean, numeric, or temporal types right now
+  # TODO: remove when ARROW-12955 merges
+  # If true/false are R types, we can use `is.*` directly
```

Comment:
@thisisnic may I push some commits to this PR that make the `is.*` functions compatible with R scalars so that this can be simplified? I've been meaning to do that in ARROW-13118 anyhow.

Reply:
feel free, nic is off through monday so I took this over — simplifying it would be great though!

Reply:
@ianmcook I question whether this is. checking is desirable at all here